File: C++/83. Remove Duplicates from Sorted List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        ListNode* p = head;
        while(p && p->next != nullptr)
        {
            int v1 = p->val;
            int v2 = p->next->val;
            if(v1==v2)
                p->next = p->next->next;
            else
                p = p->next;
        }
        return head;
    }
};

File: Python/83. Remove Duplicates from Sorted List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        p = head
        while p != None and p.next != None:
            v1, v2 = p.val, p.next.val
            if v1 == v2:
                p.next = p.next.next
            else:
                p = p.next
        return head